src/gen9_avc_encoder.c

```diff
@@ -3926,55 +4442,378 @@ gen9_avc_send_surface_mbenc(VADriverContextP ctx,
     return;
 }
 
-static VAStatus
-gen9_avc_kernel_mbenc(VADriverContextP ctx,
-                      struct encode_state *encode_state,
-                      struct intel_encoder_context *encoder_context,
-                      bool i_frame_dist_in_use)
+static void
+gen9_avc_fei_send_surface_mbenc(VADriverContextP ctx,
+                                struct encode_state *encode_state,
+                                struct i965_gpe_context *gpe_context,
+                                struct intel_encoder_context *encoder_context,
+                                void * param_mbenc)
 {
     struct i965_driver_data *i965 = i965_driver_data(ctx);
-    struct i965_gpe_table *gpe = &i965->gpe_table;
     struct encoder_vme_mfc_context * vme_context = (struct encoder_vme_mfc_context *)encoder_context->vme_context;
-    struct generic_encoder_context * generic_ctx = (struct generic_encoder_context *)vme_context->generic_enc_ctx;
-    struct i965_avc_encoder_context * avc_ctx = (struct i965_avc_encoder_context *)vme_context->private_enc_ctx;
     struct generic_enc_codec_state * generic_state = (struct generic_enc_codec_state *)vme_context->generic_enc_state;
     struct avc_enc_state * avc_state = (struct avc_enc_state *)vme_context->private_enc_state;
+    VAEncSliceParameterBufferH264 * slice_param = avc_state->slice_param[0];
+    VAEncMiscParameterFEIFrameControlH264 *fei_param = NULL;
+    struct object_buffer *obj_buffer = NULL;
+    struct buffer_store *buffer_store = NULL;
+    struct object_surface *obj_surface = NULL;
+    struct gen9_surface_avc *avc_priv_surface;
+    struct i965_gpe_resource *gpe_resource;
```

Comment:
avc_private_surface never get an unknown value because:
 if (!obj_surface || !obj_surface->private_data)
        return;

gpe_resource will always get the known value since it is assigned with an address of the already defined memory in avc_private_surface.